src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -0,0 +1,207 @@
+	"golang.org/x/sync/errgroup"
+)
+
+func newMovePartitionReplicasCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		ns         string
+		partitions []string
+		topic      string
+	)
+	cmd := &cobra.Command{
+		Use:   "move",
+		Short: "Move partition replicas across nodes / cores",
+		Long:  helpAlterAssignments,
+		Run: func(cmd *cobra.Command, topics []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			g, egCtx := errgroup.WithContext(cmd.Context())
+			for _, partition := range partitions {
+				partition := partition
+				g.Go(func() (rerr error) {
+					if len(topics) > 0 { // foo -p 0:1,2,3
+						for _, t := range topics {
+							_, _, part, err := extractNTP(t, partition)
+							out.MaybeDie(err, "failed to extract topic/partition: %s\n", err)
+
+							if nt := strings.Split(t, "/"); len(nt) == 1 {
+								ns = "kafka"
+								topic = nt[0]
+							} else {
+								ns = nt[0]
+								topic = nt[1]
+							}
+
+							current, err := cl.GetPartition(egCtx, ns, topic, part)
+							out.MaybeDie(err, "unable to get partition: %s\n", err)
+
+							newReplicas, err := configureReplicas(egCtx, cl, partition, current.Replicas)
+							out.MaybeDie(err, "unable to configure new replicas: %v\n", err)
+
+							err = cl.MoveReplicas(egCtx, ns, topic, part, newReplicas)
+							out.MaybeDie(err, "unable to move partition replicas: %v\n", err)
+						}
+					} else { // -p foo/0:1,2,3
+						ns, topic, part, err := extractNTP("", partition)
+						out.MaybeDie(err, "failed to extract topic/partition: %s\n", err)
+
+						current, err := cl.GetPartition(egCtx, ns, topic, part)
+						out.MaybeDie(err, "unable to get partition: %s\n", err)
+
+						newReplicas, err := configureReplicas(egCtx, cl, partition, current.Replicas)
+						out.MaybeDie(err, "unable to configure new replicas: %v\n", err)
+
+						err = cl.MoveReplicas(egCtx, ns, topic, part, newReplicas)
+						out.MaybeDie(err, "unable to move partition replicas: %v\n", err)
+					}
+					return nil
+				})
+			}
+
+			if err := g.Wait(); err != nil {
+				fmt.Printf("failed to begin movements: %v\n", err)
+			}
+
+			fmt.Println("Successfully began partition movements.\n\nCheck the movement status with 'rpk cluster partitions move-status' or see new assignments with 'rpk topic describe -p TOPIC'.")
+		},
+	}
+	cmd.Flags().StringArrayVarP(&partitions, "partition", "p", nil, "Topic-partitions to move and new replica locations (repeatable)")
+	cmd.MarkFlagRequired("partition")
+	return cmd
+}
+
+// extractNTP parses the partition flag with format; foo/0:1,2,3 or 0:1,2,3.
+// It extracts letters before the colon and formats it.
+func extractNTP(topic string, ntp string) (ns string, tp string, part int, err error) {
+	re := regexp.MustCompile(`^((?:[^:]+/)?\d+):.*$`)
```

Comment:
Also I think since rpk is on go 1.21, this can be simplified to just:
```go
var ntpRe = sync.OnceValue(func() *regexp.Regexp) {
    return regexp.MustCompile(...)
})
```
and then usage is just `ntpRe().FindStringSubmatch(...)`

Reply:
rpk uses go 1.20, right? https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/go.mod#L3

Reply:
I thought we just bumped this ... you're right, it's currently 1.20.